Sources/Resource/Resource.swift

```diff
@@ -4,25 +4,30 @@ public protocol ResourceDecoder {
     func decode<Value: Decodable>(_ type: Value.Type, from data: Data?, response: HTTPURLResponse) throws -> Value
 }
 
+public enum ResourceDecoding {
+    case background(ResourceDecoder)
```

Comment:
What do you think of `default` or similar? I'm having a little semantic trouble with associating the word ResourceDecoding with those cases, they seem "different".

Reply:
`ResourceDecodingExecution` ?